core/include/scipp/core/element/to_unit.h

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include <cmath>
+#include <limits>
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/time_point.h"
+#include "scipp/core/transform_common.h"
+
+namespace scipp::core::element {
+
+namespace {
+
+// Source: https://stackoverflow.com/a/26584177/1458281
+template <class T, class U> T safe_cast(const U x) {
+  if (std::isnan(x))
+    return std::numeric_limits<T>::min(); // behavior as numpy
+  int exp;
+  std::frexp(x, &exp);
+  if (std::isfinite(x) && exp <= 8 * static_cast<int>(sizeof(T)) - 1)
+    return x;
+  return std::signbit(x) ? std::numeric_limits<T>::min()
+                         : std::numeric_limits<T>::max();
+}
+
+template <class T>
+constexpr auto round = [](const auto x) {
+  if constexpr (std::is_integral_v<T>)
```

Comment:
It's a minor point, but we've been using `numeric` as wrapper for this type of type checking. i.e. https://github.com/scipp/scipp/blob/main/common/include/scipp/common/numeric.h#L76-L79 would it be better to have a `numeric::round`?

Reply:
Sure, once we use this more we should. At this point I'd like to get this ready so @nvaytet can use it in his PR.